ha/setup/post_disruptive_upgrade.py

```diff
@@ -0,0 +1,163 @@
+# cortx-questions@seagate.com.
+
+"""Module which performs post-upgrade routines for the Disruptive Upgrade feature."""
+
+import os
+from shutil import copystat, copyfile
+from xml.etree import ElementTree
+from xml.etree.ElementTree import Element
+
+from cortx.utils.log import Log
+import yaml
+from ha.core.error import UpgradeError
+from ha.execute import SimpleCommand
+from ha.const import (BACKUP_DEST_DIR_CONF, CONFIG_DIR, SOURCE_CONFIG_PATH,
+                        RA_LOG_DIR, CHECK_PCS_STANDBY_MODE, \
+                        CLUSTER_STANDBY_UNSTANDBY_TIMEOUT, PCS_CLUSTER_STANDBY, \
+                        PCS_CLUSTER_UNSTANDBY)
+from ha.setup.create_pacemaker_resources import create_all_resources
+
+
+def _get_cib_xml() -> Element:
+    """Call `pcs cluster cib` and return XML object for further parsing."""
+    output, _, _ = SimpleCommand().run_cmd("pcs cluster cib")
+    return ElementTree.fromstring(output)
+
+def _check_for_any_resource_presence() -> None:
+    '''Check if any resources are already present in a cluster.
+       if yes, means, pre-upgrade steps failed. hence exit'''
+
+    Log.info('Check for any resource presence in a cluster')
+
+    root = _get_cib_xml()
+    resource_list = [e.attrib["id"] for e in root.findall(".//lrm_resource")
+                if "id" in e.attrib]
+
+    if resource_list:
+        raise UpgradeError('Some resources are already present in the cluster. \
+                            Perform Upgrade process again')
+
+
+def _is_cluster_standby_on() -> None:
+    '''Check if cluster is in standby mode. If not, make standby mode ON'''
+
+    Log.info('Check cluster is in standby mode')
+    value = SimpleCommand().run_cmd(CHECK_PCS_STANDBY_MODE)
+
+    standby_value = value[0].split(' ')[3].strip('\n').split('=')
+
+    if standby_value[1].lower() != 'on':
+        Log.warn('cluster is not in standby mode.')
+        Log.info('switching the cluster in standby mode for performing post upgrade routines')
+        _switch_cluster_mode(PCS_CLUSTER_STANDBY)
+    Log.info('#### All post-upgrade prerequisites are in place ####')
+
+def _restore_consul_backup():
+
+    Log.info('Restoring the consul backup')
+
+def _yaml_to_dict(yaml_file=None):
+    '''
+       Convert yaml format key value info i the form of
+       python dictionary key value
+    '''
+    if yaml_file is None:
+        raise UpgradeError('yaml file path can not be None. Please provide the \
+                 HA yaml conf file path for conversion')
+    with open(yaml_file, 'r') as conf_file:
+        file_as_dict = yaml.safe_load(conf_file)
+    return file_as_dict
+
+def _load_config() -> None:
+    '''
+       Load the new config at proper location after the
+       RPM upgrade as part of post-upgrade process
+    '''
+
+    dest_dir = CONFIG_DIR
+    new_src_dir = SOURCE_CONFIG_PATH
+
+    HA_BACKUP_CONF_FILE = BACKUP_DEST_DIR_CONF + '/' + 'ha.conf'
+    HA_SOURCE_CONF = SOURCE_CONFIG_PATH + '/' + 'ha.conf'
```

Comment:
Please do not hardcode these files. Pass these as argument to perform_post_upgrade function and keep the default value to the arguments as above.

Reply:
Ok. Done.